backend/src/api/components/friend/controller.ts

```diff
@@ -0,0 +1,51 @@
+import { IncomingMessage } from 'http';
+import { Friend } from './model/Friend';
+import { Friends } from './model/Friends';
+import fs from "fs";
+
+export class FriendController {
+
+    createNewFriend(friendData: FriendDTO): Promise<Friend> {
+        const newFriend = new Friends().create(friendData.name);
+
+        // store friend to a file
+        return new Promise((resolve, reject) => {
+            fs.readFile("store/friends.json", "utf8", (err, data) => {
```

Comment:
Try to pull all of these file reading details into a separate function and reuse that. Maybe you can get object oriented with it and think about your store(s) as an object which you can retrieve things from?

Reply:
Refactored this bit and created a Store object